### PR TITLE
New initcap / initcapUTF8 functions (Capital/Title Case)

### LINKED ISSUE #45217: INITCAP function (Capital/Title Case)
oracle: https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/INITCAP.html
snowflake: https://docs.snowflake.com/en/sql-reference/functions/initcap.html
postgres: https://www.postgresql.org/docs/9.1/functions-string.html
informix: https://www.ibm.com/docs/en/informix-servers/12.10?topic=smf-initcap-function

and also InitCapUTF8?

### PR DESCRIPTION
**Pull request by kardymonds**
Close https://github.com/ClickHouse/ClickHouse/issues/45217

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added new initcap / initcapUTF8 functions which convert the first letter of each word to upper case and the rest to lower case.

### Documentation entry for user-facing changes

- [ V] Documentation is written (mandatory for new features)

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit fe1cf294fb9fd239d449b4ed464bf7e1c3e2c207 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51735/fe1cf294fb9fd239d449b4ed464bf7e1c3e2c207/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟡 pending</td></tr>
</table>

File: src/Functions/initcap.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#include <Functions/FunctionFactory.h>
```
Comment (kardymonds) [inline src/Functions/initcap.cpp:1]: Basically I copied from [LowerUpperImpl.h](https://raw.githubusercontent.com/ClickHouse/ClickHouse/master/src/Functions/LowerUpperImpl.h) and [LowerUpperUTF8Impl.h](https://raw.githubusercontent.com/ClickHouse/ClickHouse/master/src/Functions/LowerUpperUTF8Impl.h)

**Comment by kardymonds** [comment]
@vdimir Passed the tests. Is there anything else i need to do?

**Comment by vdimir** [comment]
> Passed the tests. Is there anything else i need to do?

@kardymonds I went trough the PR, and it looks good. I'll read the code once more a bit deeper soon, and will leave the comments if any. But I suppose it will be none or maybe only minor ones. Thank you for the PR!

**Review by vdimir — APPROVED** [review]
Thank you for adding the feature!

**Pull request merged**